core/controllers/editor.py

```diff
@@ -708,21 +1073,49 @@ def post(self, unused_exploration_id):
 class EditorAutosaveHandler(ExplorationHandler):
     """Handles requests from the editor for draft autosave."""
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': {
+                'type': 'basestring'
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'PUT': {
+            'version': {
+                'schema': {
+                    'type': 'int'
+                }
+            },
+            'change_list': {
+                'schema': {
+                    'type': 'list',
+                    'items': {
+                        'type': 'object_dict',
+                        'validation_method': (
+                            objects_validator.validate_exploration_change)
+                    }
+                }
+            }
+        },
+        'POST': {},
+        # Below two methods are not defined in handler class but they must be
+        # present in schema since these two are inherited from its parent class.
+        'GET': {},
+        'DELETE': {}
```

Comment:
In that case shouldn't the schema also be inherited(or the same) otherwise the validation won't occur.

Reply:
Yes HANDLER_ARGS_SCHEMAS is being inherited but overridden in the class. The reason I have to include these two names because otherwise it will result in backend test failure [here](https://github.com/oppia/oppia/blob/b1846a839bfb72a5430031fcd2e27638004ad92b/core/controllers/base_test.py#L1576)